### DISCUSSION TITLE
Checkout branch with discard commit

### DISCUSSION THREAD

**Discussion by fari-99** [Kong/insomnia#7568] [Ideas] [answered]
soo, my current master project is using version `9.2.0` and i am using `9.3.0 beta`, because i have feature of `postRequestScript` it changed the branch to ready to commit. but i don't want to commit/push anything, so i discard any changes, and checkout to my branch, but insomnia automatically add those `postRequestScript`, and give error "Please commit current changes before switching branches".

please add a "checkout with discard commit" options. thank you

**Comment by ihexxa** [comment, marked as answer]
Hello @fari-99,
It is fixed in #7556, you could try the beta.6 just released and see if it works for you.

**Reply by fari-99** [reply]
oh ok i will try beta.6 then. will report later.

edit: hey it works! thanks~
